drivers/sensor/mcp9808/mcp9808.c

```diff
@@ -35,6 +35,39 @@ int mcp9808_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
 	return rc;
 }
 
+int mcp9808_reg_write_16bit(const struct device *dev, uint8_t reg,
+			    uint16_t val)
+{
+	const struct mcp9808_data *data = dev->data;
+	const struct mcp9808_config *cfg = dev->config;
+	uint8_t buf[3] = {
+		reg,
+		val >> 8,       /* big-endian register storage */
```

Comment:
use facilities in `byteorder.h`

Reply:
`const` removed from data.

I'm struggling to figure out why using facilities from `byteorder.h` will make this small section any better. It's already clear what the section is doing.

I've read through 'byteroder.h' and even after changing the order, I would still need to manually place the different bytes within the array.

Reply:
you can use `sys_put_be16` to insert the value

Reply:
Sorry, but I'm still not seeing this. I can use the following to include `sys_put_be16`, but I still don't see it as being any better:

`uint8_t buff[3] = {reg};
sys_put_be16(val, &buf[1]);`

It compiles and runs, but I'm not seeing the benefit.

Reply:
Or

`uint8_t buff[3];
buff[0] = reg;
sys_put_be16(val, &buf[1]);`

Reply:
I think it is better to use system facilities when available, they'll usually be tested and in some cases optimized.

Reply:
Updated to use `sys_put_be16`